### DISCUSSION TITLE
Javascript Copy to clipboard functionality

### DISCUSSION THREAD

**Discussion by kelemenattila** [cefsharp/CefSharp#4779] [General]
I need to copy custom formatted data in javascript to clipboard using ChromiumBrowser. 
I'm adding the text to the clipboard with the following JS code:

navigator.clipboard.writeText(formattedText);

In case of ChromiumBrowser the text doesn't appear in clipboard. In case of other browsers (Chrome, Brave, Edge) works. 
I got a workaround with creating a ghost element copy content with document.execCommand('copy'), works but 1/10 doesn't only if I debug the JS code step by step. By the way execCommand is obsolete :( 

I attached the full workarround code but I prefer to use somehow the navigator to copy to clipboard if it is possible.
Thanks :) 

const element = document.createElement('textarea');
element.value = formattedText;
 document.body.appendChild(element); 
element.select();
 document.execCommand('copy'); 
document.body.removeChild(element);

**Comment by amaitland** [comment]
> In case of ChromiumBrowser the text doesn't appear in clipboard

By default permission to cliipboard should be disabled. You can grant permissions programmatically.

- https://github.com/cefsharp/CefSharp/blob/800cbcf0fd58d85bfa0c3f57636ddfa823a1b830/CefSharp/IWebBrowser.cs#L140
- https://github.com/cefsharp/CefSharp/blob/800cbcf0fd58d85bfa0c3f57636ddfa823a1b830/CefSharp.Example/Handlers/ExamplePermissionHandler.cs#L13

```c#
chromiumWebBrowser.PermissionHandler = new MyPermissionHandler();`
```

**Reply by kelemenattila** [reply]
Thanks. It works
